### PR TITLE
Respect comments in SQL + move code to extensions-base

### LINKED ISSUE #8221: DBT with Nessie: ParseException
### Issue description

Hello,

Running DBT with Nessie as follows results in:
```
    == SQL ==
    /* {"app": "dbt", "dbt_version": "1.7.9", "profile_name": "spark", "target_name": "prod", "node_id": "model.spark.sandbox__iceberg_table"} */

            CREATE BRANCH IF NOT EXISTS test IN iceberg_catalog
    ---------------^^^

    	at org.apache.spark.sql.catalyst.parser.ParseException.withCommand(parsers.scala:257)
    	at org.apache.spark.sql.catalyst.parser.AbstractParser.parse(parsers.scala:98)
    	at org.apache.spark.sql.execution.SparkSqlParser.parse(SparkSqlParser.scala:54)
    	at org.apache.spark.sql.catalyst.parser.AbstractSqlParser.parsePlan(AbstractSqlParser.scala:68)
    	at org.apache.spark.sql.catalyst.parser.extensions.NessieSparkSqlExtensionsParser.parsePlan(NessieSparkSqlExtensionsParser.scala:114)
    	at org.apache.spark.sql.catalyst.parser.extensions.IcebergSparkSqlExtensionsParser.parsePlan(IcebergSparkSqlExtensionsParser.scala:126)
    	at org.apache.spark.sql.SparkSession.$anonfun$sql$5(SparkSession.scala:684)
    	at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:138)
    	at org.apache.spark.sql.SparkSession.$anonfun$sql$4(SparkSession.scala:683)
    	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
    	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:682)
    	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:713)
    	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:744)
    	at org.apache.spark.sql.SQLContext.sql(SQLContext.scala:651)
    	at org.apache.spark.sql.hive.thriftserver.SparkExecuteStatementOperation.org$apache$spark$sql$hive$thriftserver$SparkExecuteStatementOperation$$execute(SparkExecuteStatementOperation.scala:227)
    	... 16 more
```

Found this and indeed solved it for me: [link](
https://stackoverflow.com/questions/76581576/syntax-error-when-using-nessie-commands-with-dbt-but-not-using-spark)

Setup:
- Spark 3.5.1
- Nessie 0.79.0
- Iceberg 1.5.1
- DBT 1.7.9
- `dbt_project.yml`:
   ```
  pre-hook:
    - 'CREATE BRANCH IF NOT EXISTS test IN iceberg_catalog;'
    - 'USE REFERENCE test IN iceberg_catalog;'
   ```

Reproduce:
Execute:
```sql
/* {"app": "dbt", "dbt_version": "1.7.9", "profile_name": "spark", "target_name": "prod", "node_id": "model.spark.sandbox__iceberg_table"} */

CREATE BRANCH IF NOT EXISTS test IN iceberg_catalog
```

I'm not even sure it's a bug yet (would need to dig deeper) but wanted to share in case someone bumps into it.

Bests,

### PR DESCRIPTION
**Pull request by snazy**
The issue described in #8221 is in `NessieSparkSqlExtensionsParser#isNessieCommand`, which does not respect SQL comments. The actual fix for that issue is in there. Also slightly updated the antlr grammar.

The rest of this PR is "boring" - unifies the `Base*Exec` into the `*Exec` ones and to the `:nessie_spark-extendions-base` package, leaving just the integration-test hull in the projects for the different Spark versions.

Fixes #8221

### CODE REVIEW

File: integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/NessieSparkSqlExtensionsParser.scala [hunk 1]
```diff
@@ -116,7 +116,17 @@ class NessieSparkSqlExtensionsParser(delegate: ParserInterface)
   }
 
   private def isNessieCommand(sqlText: String): Boolean = {
-    val normalized = sqlText.toLowerCase(Locale.ROOT).trim()
+    val normalized = sqlText
```
Comment (adutra) [inline integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/NessieSparkSqlExtensionsParser.scala:119]: It looks frustrating to have to use regex in order to "pre-parse" something that can be parsed with a proper Antlr grammar. Shouldn't we get rid of `isNessieCommand`?

Comment (adutra) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/NessieSparkSqlExtensionsParser.scala:119]: BTW I could find at least one statement that is correctly parsed by the grammar, but fails to be parsed by the regexes:

```
/* leading /* leading */ */ CREATE BRANCH IF NOT EXISTS %s_other IN nessie
```

File: integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java [hunk 2]
```diff
@@ -90,6 +90,22 @@ public void backquotedRefName(String branchName) throws NessieNotFoundException
     assertThat(sql("DROP BRANCH `%s` IN nessie", branchName)).hasSize(1).containsExactly(row("OK"));
   }
 
+  @ParameterizedTest
+  @ValueSource(
+      strings = {
+        "USE REFERENCE %s IN nessie",
+        "/* leading */ CREATE BRANCH IF NOT EXISTS %s_other IN nessie",
```
Comment (adutra) [inline integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java:97]: A few more interesting test cases (but the first new one will fail):

```suggestion
        "/* leading */ CREATE BRANCH IF NOT EXISTS %s_other IN nessie",
        "/* leading /* leading */ */ CREATE BRANCH IF NOT EXISTS %s_other IN nessie",
        "/* leading -- leading \n */ CREATE BRANCH IF NOT EXISTS %s_other IN nessie",
        "-- leading \n CREATE BRANCH IF NOT EXISTS %s_other IN nessie",
        " -- leading \n -- leading \n -- leading \n CREATE BRANCH IF NOT EXISTS %s_other IN nessie",
```

File: integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/NessieSparkSqlExtensionsParser.scala (hunk 1 above)
Comment (snazy) [reply integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/NessieSparkSqlExtensionsParser.scala:119]: It's the same approach as in Iceberg

File: integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java (hunk 2 above)
Comment (snazy) [reply integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java:97]: added

**Review by adutra — APPROVED** [review]

**Pull request merged**